File: backend/src/main/java/com/documed/backend/schedules/TimeSlot.java
package com.documed.backend.schedules;

import com.documed.backend.users.User;
import java.time.LocalTime;
import java.util.Date;
import lombok.AllArgsConstructor;

@AllArgsConstructor
public class TimeSlot {
  private int id;
  private final User doctor;
  private final LocalTime startTime;
  private final LocalTime endTime;
  private final Date date;
  private boolean isBusy;
}

File: frontend/src/shared/types/TimeSlot.ts
import { UserLite } from './User';

export interface TimeSlot extends TimeSlotLite {
  doctor: UserLite;
}

export interface TimeSlotLite {
  id: number;
  startTime: string;
  endTime: string;
  date: Date;
  isBusy: boolean;
}
